src/sfctl/tests/help_text_test.py

```diff
@@ -0,0 +1,375 @@
+                                                                  commands_index)
+                subgroups_index = updated_indices[0]
+                commands_index = updated_indices[1]
+
+            # If section is still 'Start', the something has gone wrong.
+            # It means that lines were not processed
+            # correctly, since we expect some sections to appear.
+            self.assertNotEqual('Start', section,
+                                msg='Command {0} has incomplete help text: {1}'.format(help_command, returned_string))  # pylint: disable=line-too-long
+
+            # Check that we have traversed completely through both
+            # subgroups and commands
+            self.assertEqual(len(commands), commands_index,
+                             msg=('Not all commands listed in help text for '
+                                  + help_command
+                                  + '. This may be a problem due incorrect expected ordering.'))
+            self.assertEqual(len(subgroups), subgroups_index,
+                             msg=('Not all subgroups listed in help text for '
+                                  + help_command
+                                  + '. This may be a problem due incorrect expected ordering.'))
+
+            print(file=stderr)
+
+        except Exception as e:  # pylint: disable=broad-except
+            if not err:
+                self.fail(msg='ERROR: Command {0} returned error at execution. Output: {1} Error: {2}'.format(help_command, returned_string, str(e)))  # pylint: disable=line-too-long
+            else:
+                self.fail(msg='ERROR: Command {0} returned error at execution. Output: {1} Error: {2}'.format(help_command, returned_string, err))  # pylint: disable=line-too-long
+
+        # Once validation is done for the provided command_input,
+        # if there are any commands returned in the help text, validate those commands.
+        for command in commands:
+            self.validate_output(command_input + ' ' + command)
+
+    def test_help_documentation(self):
+        """ Tests all help documentation to ensure that all commands have help text.
+        This does not test for typos / correctness in the text itself.
+        This test calls validate_output on all commands which sfctl has, without the
+        '-h' flag included. The flag will be added by validate_ouput.
+
+        Note: validate_output expects subgroups and commands in order. If out of alphabetical
+        order, you will see an error for not all commands/subgroups being listed.
+
+        Note: you do not need to call individual commands. Commands listed in the
+        'commands' list will be called and verified automatically. You DO need
+        an entry for each subgroup."""
+
+        self.validate_output(
+            'sfctl',
+            subgroups=('application', 'chaos', 'cluster', 'compose', 'is', 'node',
```

Comment:
I think we should maybe ask for this functionality from the knack module. Giving a loaded CLI, it would be useful to return a list of all possible commands.